fairlearn/datasets/_fetch_data.py

```diff
@@ -0,0 +1,91 @@
+import warnings
+from ._common import FairnessWarning
+
+from sklearn.datasets import fetch_openml
+
+
+def fetch_boston(*, version='1', cache=True, data_home=None,
+                 as_frame=False, return_X_y=False, warn=True):
+    """Load the boston housing dataset (regression).
+
+    Download it if necessary.
+
+    ==============   ==============
+    Samples total               506
+    Dimensionality               13
+    Features                   real
+    Target           real 5. - 50.
+    ==============   ==============
+
+    This dataset has known fairness issues. There's a "lower status of
+    population" (LSTAT) parameter that you need to look out for and a column
+    that is a derived from the proportion of people with a black skin color
+    that live in a neighborhood (B). This dataset should be used to explore
+    and benchmark fairness. Predicting a house-price while ignoring these
+    sensitive attributes is naive.
+
+    Parameters
```

Comment:
Just a note that this is `numpydoc` which I very much like, but it's not the same as the rest of the repo (also cc @romanlutz)

Reply:
I copied scikit-learn here, if this project prefers a different standard I'll gladly hear it. Sphinx seemed to render it fine tho.

Reply:
The docstring standard is currently being debated (#314) and we might end up switching to `numpydoc`. I'm not sure it makes sense to rewrite the docstring if it renders correctly on the webpage--unless #314 is resolved in favor of status quo by the time this PR gets in.

Reply:
I like @adrinjalali 's proposal of just switching gradually. That would mean you can leave this in numpydoc format. FYI this is what it looks like in the generated documentation: https://250-133444044-gh.circle-artifacts.com/0/docs/_build/html/api_reference/fairlearn.datasets.html 
Is that what you would have expected? Would be nice if the feature/column names showed up.

Reply:
@romanlutz that's a nice point, datasets have an opportunity to render the description in the documentation. 

Here's what it looks like using base sphinx. 

![image](https://user-images.githubusercontent.com/1019791/83940974-e638fd80-a7e7-11ea-808a-6703996b5e45.png)

Would this suffice?

Reply:
This is great!